Aptoide/aptoide-client-v8
# Feature/asv 639 ab testing implementation

**What does this PR do?**

This PR aims to implement an extensible and reusable AB Testing feature; This involved creating everything from the service to the interactor;

**Database changed?**

   Yes 

**Where should the reviewer start?**

- [x] ABTestManager.java
- [x] ABTestCenterRepository.java

**How should this be manually tested?**

You'll need to check the if, on install press, the requests to assignment, impression, and, on click on continue/share, action; You should also verify the local cache by clicking install again without closing the app (you should see the experiment, make impression and action clicks but not the assignment) and the persistence by killing the app and pressing install again (it should act the same as with the local cache in this instance);

**What are the relevant tickets?**

  Tickets related to this pull-request: [ASV-639](<https://aptoide.atlassian.net/browse/ASV-639>)


**Code Review Checklist**
